### PR TITLE
handle single subscription

### PR DESCRIPTION
**Pull request by nachozt**
*Related Defect(s), Issue(s) or Task(s)*

https://github.com/Zetatango/zetatango/issues/6029

*Why?*

The way Hopper works in our Rails apps is by adding an initializer that runs 3 operations:
```
1- Hopper.init_channel(..)
2- Require any file that has a Hopper subscription
3- Hopper.start_listening
```

Step 2 was added in case the app is lazy loaded and it didn't set the subscription on time (before start_listening). The issue can also be addressed by eager loading our apps, but we don't want to force them to do that when it's not needed. Also, we shouldn't rely on that, we better make sure none subscribers are missed. 

So we'll leave our Rails apps as they are now, and make Hopper handle the single subscription for the same subscriber.

*How?*

Check if the same subscriber was already added to the list of subscribers and skip it if that's the case.

*Risks*

NA

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,25 @@
+exit
```
Comment (bcarr092) [inline .byebug_history]: Added by mistake I assume

File: lib/hopper.rb [hunk 2]
```diff
@@ -33,6 +33,8 @@ def publish(message, key)
     end
 
     def add_subscriber(subscriber)
+      return if subscribers.include? subscriber
```
Comment (bcarr092) [inline lib/hopper.rb]: We generally use parenthesis on method calls

File: .byebug_history (hunk 1 above)
Comment (dragoszt) [reply .byebug_history]: Add it to .gitignore

File: lib/hopper.rb (hunk 2 above)
Comment (dragoszt) [inline lib/hopper.rb]: How is the subscriber uniquely identified? Don't we need to override the equal? method on the subscriber class to identify a subscriber by the class+event+method tuple?

**Pull request merged**